source/steamlib/steamlib_main.cpp

```diff
@@ -16,71 +16,202 @@ You should have received a copy of the GNU Lesser General Public
 License along with this library.
 */
 
+#include "../gameshared/q_shared.h"
+#include "../qcommon/qcommon.h"
 #include "steamlib_local.h"
+#include <steam/steam_api.h>
 
-namespace WSWSTEAM {
+namespace WSWSTEAM
+{
+typedef void ( *auth_callback_t )( void *, size_t );
+
+struct CSteamCallbacks {
+	auth_callback_t m_callback;
+	uint8 m_AuthTicket[1024];
+	uint32 m_AuthTicketLength;
+	HAuthTicket m_AuthTicketResult;
+
+	CSteamCallbacks() : m_callback( NULL ), m_AuthTicketLength( 0 ), m_AuthTicketResult( k_HAuthTicketInvalid ) {}
+
+	int GetAuthSessionTicket( void ( *callback )( void *, size_t ) )
+	{
+		m_callback = callback;
+		m_AuthTicketResult = SteamUser()->GetAuthSessionTicket( m_AuthTicket, sizeof( m_AuthTicket ), &m_AuthTicketLength );
+		return m_AuthTicketResult != k_HAuthTicketInvalid ? 1 : 0;
+	}
+
+	STEAM_CALLBACK( CSteamCallbacks, OnGetAuthSessionTicketResponse, GetAuthSessionTicketResponse_t );
+	STEAM_CALLBACK( CSteamCallbacks, OnWorkshopItemInstalled, ItemInstalled_t );
+
+	STEAM_CALLBACK( CSteamCallbacks, OnGameServerChangeRequested, GameServerChangeRequested_t);
+};
+static CSteamCallbacks *g_pCallbacks = nullptr;
+
+void CSteamCallbacks::OnGetAuthSessionTicketResponse( GetAuthSessionTicketResponse_t *pParam )
+{
+	if( pParam->m_eResult != k_EResultOK )
+		m_AuthTicketLength = 0;
+
+	if (m_callback)
+	{
+		m_callback(m_AuthTicket, m_AuthTicketLength);
+	}
+}
+
+void CSteamCallbacks::OnWorkshopItemInstalled( ItemInstalled_t *pParam )
+{
+	if( pParam->m_unAppID == SteamUtils()->GetAppID() ) {
+		Com_Printf( "New workshop item installed, please restart!\n" );
```

Comment:
I did not see a good way to reload the maplist / filesystem, so we could probably even remove this callback.